### ISSUE TITLE
Support sudo within recipe

### ISSUE THREAD

**Issue by cgspeck** [hamidnazari/amibaker#22]
It would be great if the recipe -> task -> run could be extended to support executing scripts as sudo on the remote end. This would allow people to write src scripts without having to put sudos everywhere. 

It could be implemented by adding an optional key to a run task, e.g.:

```
  - run:
    - src: /path/to/local/script
      dest: /path/to/remote/dir
      cwd: /path/to/another/remote/dir
      sudo: True
  - run:
    - cwd: /path/to//remote/dir
      body: |
       sudo yum update -y
       sudo yum install -y telnet
      sudo: True
```

Fabric already supports this. It would require changing logic of `Provisioner._run` constructor and approx line 90 within the `try...except...finally` block.

**Comment by hamidnazari** [comment]
Would be nice to add the same key to `copy` tasks. At the moment it uses sudo by default to lazily avoid write permissions.

**Comment by hamidnazari** [comment]
I've commenced working on this.

**Comment by hamidnazari** [comment]
Implemented in #24.
